src/mechanics/integrands/MomentumBalance.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include "mechanics/integrands/TimeDependent.h"
+#include "mechanics/constitutive/laws/LinearElastic.h"
+#include "mechanics/nodes/DofType.h"
+#include "mechanics/interpolation/TypeDefs.h"
+#include "mechanics/cell/CellIpData.h"
+#include "mechanics/cell/CellData.h"
+
+namespace NuTo
+{
+namespace Integrand
+{
+namespace TimeDependent
+{
+
+template <int TDim>
+class MomentumBalance : public Interface
+{
+public:
+    MomentumBalance(const NuTo::DofType& dofType, const Laws::MechanicsInterface<TDim>& law)
```

Comment:
const of the law would IMO not allow to perform an update of the history data

Reply:
That somehow depends on where the history data is stored. I have not thought about that when I implemented the code above. 
- store at law: We somehow have to separate material parameters from history data to avoid duplicate storage of material parameters. 
- store at integrand: We somehow lose the type of the history data. 

So I would prefer your suggestion to store it at the law and make that nonconst. Actually, the integrand then _owns_ the law and we need a `unique_ptr` instead of a reference.

Reply:
Btw: You can have a look/edit at the [Project PDE](https://github.com/nutofem/nuto/projects/1). I added your remark there.

Reply:
I thought that the history data is part of the integrand there is (at least in the standard case) no law. The only thing I would use is a reference to a list of material parameters (or similar a class that stores these values such as a class MaterialParameters::LinearElastic. Why do we loose the type here, since the Stress routine of the law is virtual and in the derived class (that stores the history data) this type is known.

Reply:
Those were two options. We lose the type information, if we store the history data of the law at the integrand. If we store it at the law, there is no problem. 
One alternative to a MaterialParameters class is to use our current LawWithHistoryData pattern. So the laws would derive from `MechanicsInterfaceWithData`. 
~~~cpp
template <typename TLaw>
struct MechanicsInterfaceWithData : MechanicsInterface
{
    TLaw::HistoryDataType mHistoryData;

    EngStress Stress(Strain, deltaT) override
    {
        return Stress(Strain, mHistoryData, deltaA);
    }

protected:
    virtual EngStress Stress(Strain, TLaw::HistoryDataType&, deltaT) = 0;
    // overridden by actual laws.
}
~~~
Your reference approach seems a lot easier. The usage will not be as nice.
~~~cpp
LawMaterialParameters params(E, nu); // must not go out of scope
Law law(params);
// vs
Law law(E, nu);
~~~
Everything is solveable. I think we get a feel for a nice solution if we start implementing some laws with data.